great_expectations/rule_based_profiler/attributed_resolved_metrics.py

```diff
@@ -53,6 +53,14 @@ def _detect_illegal_array_type_or_shape(values: MetricValues) -> bool:
         return False
 
     if _detect_illegal_array_type_or_shape(values=metric_values):
+        value: MetricValue
+        if (
+            metric_values is not None
+            and isinstance(metric_values, (list, tuple))
```

Comment:
This should render #5927 moot, but for corner cases, is it possible we still want to also handle it downstream?

Reply:
@NathanFarmer I am not sure -- I was hoping that once this is merged you would make best effort assessment and explain and re-issue the PR if you deem appropriate.  (Alternatively, we can reserve this higher-level check till the problem manifests itself vividly, making the development of an appropriate test a bit easier.). Thanks!